src/Bicep.LangServer.IntegrationTests/LangServerScenarioTests.cs

```diff
@@ -78,10 +82,21 @@ async Task publish(string source)
         var cacheRoot = FileHelper.GetCacheRootDirectory(TestContext);
         var helper = await MultiFileLanguageServerHelper.StartLanguageServer(
             TestContext,
-            services => services
-                .WithFeatureOverrides(new(CacheRootDirectory: cacheRoot))
-                .WithContainerRegistryClientFactory(clientFactory)
-                .AddSingleton<IModuleRestoreScheduler, ModuleRestoreScheduler>());
+            services =>
+            {
+                services = services
+                    .WithFeatureOverrides(new(CacheRootDirectory: cacheRoot))
+                    .WithContainerRegistryClientFactory(clientFactory)
+                    .AddSingleton<IModuleRestoreScheduler, ModuleRestoreScheduler>();
+
+                // Using a mock IPublicModuleIndexHttpClient since this test doesn't require public registry modules. 
+                // This prevents downloading public registry metadata on each run, avoiding potential flakiness 
+                // or test timeouts caused by slow internet connections.
```

Comment:
I discovered this after experiencing an internet outage and switching to a hotspot.